crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -3,14 +3,114 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use syn::{parse::ParseStream, Attribute, Ident, Result, Token, Type};
+use std::collections::HashSet;
+
+use syn::{
+    parse::{Error, ParseStream},
+    punctuated::Punctuated,
+    spanned::Spanned,
+    Attribute, Expr, Ident, Meta, MetaNameValue, Path, Result, Token, Type,
+};
+
+use quote::format_ident;
+
+/// An optional identifier of the functions passed with flags
+/// e.g. read = my_getter, IdentFlag would be used to store my_getter
+type IdentFlag = Option<Ident>;
+
+/// Struct for storing the flags provided for a QProperty and their optional identifiers ([IdentFlag])
+#[derive(Debug)]
+pub struct QPropertyFlags {
+    read: IdentFlag, // TODO: maybe change this to better represent the data
+    write: Option<IdentFlag>,
+    notify: Option<IdentFlag>,
+}
+
+impl QPropertyFlags {
+    pub fn new() -> Self {
+        Self {
+            read: None,
+            write: None,
+            notify: None,
+        }
+    }
+
+    pub fn all_flags() -> Self {
+        Self {
+            read: None,
+            write: Some(None),
+            notify: Some(None),
+        }
+    }
+
+    fn set_field_by_string(&mut self, key: String, value: IdentFlag) -> Result<String> {
```

Comment:
so you could instead change this to accept the `Path` as the `key`.

Then instead use our `syntax::path::path_compare_str` like this, unfortunately it means we can't use a `match` but it allows you to compare Path's as at the moment if you have `#[qproperty(T, name, read::thing)` you would consider that as `read` when it's actually an invalid path?

```write
if path_compare_str(&key /* which is of type Path */, &["read"]) {
  // we have a read
} else if path_compare_str(&key, &["write"]) {
  // we have a write
...
} else {
  // unknown we can error with the original Path span
  return Err(Error::new(key.span(), ...);
}

Reply:
Aha but as we are comparing only one level here you could use the built in method in `syn`.

There is `Path::is_ident` which you could then use like this...
https://docs.rs/syn/latest/syn/struct.Path.html#method.is_ident

```rust
if key.is_ident("read") {
  // we have a read
} else if key.is_ident("write") {
  // we have a write
...
} else {
  // unknown we can error with the original Path span
  return Err(Error::new(key.span(), ...);
}
```

Reply:
As discussed there is also `get_ident` and `require_ident` which could be useful too

Reply:
See other comment about using `require_ident`